hibernate-core/src/main/java/org/hibernate/query/internal/QueryParameterBindingImpl.java

```diff
@@ -227,6 +229,15 @@ public void setBindValues(
 		this.explicitTemporalPrecision = temporalTypePrecision;
 	}
 
+	@Override
+	public MappingModelExpressable getType() {
+		return type;
+	}
+
+	@Override
+	public void setType(MappingModelExpressable type) {
+		this.type = type;
+	}
```

Comment:
I think we could eliminate two unnecessary generics warning by making use of `<T>` above.